Source/Core/Common/Image.cpp

```diff
@@ -0,0 +1,183 @@
+// Copyright 2016 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#include <Common/Image.h>
+
+#include <csetjmp>
+#include <cstdlib>
+#include <cstring>
+#include <memory>
+#include <png.h>
+#include <vector>
+
+#include <Common/CommonTypes.h>
+#include <Common/Logging/Log.h>
+
+namespace Common
+{
+static inline int MultiplyAlpha(int alpha, int color)
+{
+  const int temp = (alpha * color) + 0x80;
+  return ((temp + (temp >> 8)) >> 8);
+}
+
+static void PremultiplyData(png_structp png, png_row_infop row_info, png_bytep data)
```

Comment:
`row_info` can be a `const png_row_info*`

Reply:
`png_set_read_user_transform_fn` doesn't want to take a function with such a signature.